pkg/event/target/mqtt.go

```diff
@@ -96,7 +101,7 @@ func (target *MQTTTarget) Send(eventData event.Event) error {
 
 	token := target.client.Publish(target.args.Topic, target.args.QoS, false, string(data))
 
-	if token.Wait() {
+	if token.WaitTimeout(150 * time.Millisecond) {
```

Comment:
If queueDir is not configured we should continue to do `token.Wait`, right?

Reply:
yes. But here the `WaitTimeout` is generic since it returns True if the context is returned before the timeout. And also in an assumption that token will respond within 150ms in a non-configured setup.

Reply:
@Praveenrajmani In a setup where `queueDir` is not configured, if the event wasn't published within 150ms, we would be returning 'success' to the user with *no acknowledgement from the MQTT broker*. This is incorrect.

Reply:
yes. got it. will change